pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -187,6 +203,38 @@ public void processSegments()
         Arrays.toString(_outputSegmentsDir.list()));
   }
 
+  private SegmentNameGenerator getSegmentNameGenerator(TableConfig tableConfig, Schema schema,
+      SegmentConfig segmentConfig) {
+    String rawTableName = TableNameBuilder.extractRawTableName(tableConfig.getTableName());
+    String segmentNameGeneratorType = segmentConfig.getSegmentNameGeneratorType();
+    if (segmentNameGeneratorType == null) {
+      segmentNameGeneratorType = SIMPLE_SEGMENT_NAME_GENERATOR;
+    }
+
+    switch (segmentNameGeneratorType) {
+      case SIMPLE_SEGMENT_NAME_GENERATOR:
+        return new SimpleSegmentNameGenerator(rawTableName, segmentConfig.getSegmentPostfix());
+      case NORMALIZED_DATE_SEGMENT_NAME_GENERATOR:
+        Preconditions.checkState(tableConfig != null,
+            "In order to use NormalizedDateSegmentNameGenerator, table config must be provided");
```

Comment:
Nit: Can be rephrased as "Table config required to use NormalizedDate..."

Reply:
Actually, this check is unnecessary. I removed the precondition block. When generating the `SegmentProcessorConfig`, there's a validation that enforces to always have `tableConfig` and `schema`.